File: app/src/main/java/dev/ashishrawat/popularmovies/database/MovieDatabase.java
package dev.ashishrawat.popularmovies.database;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import dev.ashishrawat.popularmovies.model.Movie;

@Database(entities = {Movie.class}, version = 1)
public abstract class MovieDatabase extends RoomDatabase {

    public abstract MovieDao getNoteDao();

    private static MovieDatabase movieDb;

    public static MovieDatabase getInstance(Context context) {
        if (null == movieDb) {
            movieDb = buildDatabaseInstance(context);
        }
        return movieDb;
    }

    private static MovieDatabase buildDatabaseInstance(Context context) {
        return Room.databaseBuilder(context,
                MovieDatabase.class,
                "movies")
                .allowMainThreadQueries().build();
    }

    public void cleanUp() {
        movieDb = null;
    }

}

File: app/src/main/java/dev/ashishrawat/popularmovies/database/MovieDatabase.kt
package dev.ashishrawat.popularmovies.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import dev.ashishrawat.popularmovies.model.Movie

@Database(entities = [Movie::class], version = 1)
abstract class MovieDatabase : RoomDatabase() {
    abstract val noteDao: MovieDao?
    fun cleanUp() {
        movieDb = null
    }

    companion object {
        private var movieDb: MovieDatabase? = null
        fun getInstance(context: Context): MovieDatabase? {
            if (null == movieDb) {
                movieDb = buildDatabaseInstance(context)
            }
            return movieDb
        }

        private fun buildDatabaseInstance(context: Context): MovieDatabase {
            return Room.databaseBuilder(
                context,
                MovieDatabase::class.java,
                "movies.db"
            )
                .allowMainThreadQueries().build()
        }
    }
}